### PR TITLE
add docs and use mkdocs

### PR DESCRIPTION
**Pull request by joehand**
* Adds docs/ folder and mkdocs to use rtd. 
* Updates docs for RC2

### CODE REVIEW

**Comment by karissa** [comment]
looks great!

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+## General Options
+
+* `dat --version`: get the version of dat you are running.
+* `dat --doctor`: visit the doctor to help identify connectivity issues.
+* `dat --quiet`: output only dat-link, no progress information
+* `dat --debug`: show debugging output
+
+# Local Storage
+
+When you run a command, Dat creates a hidden folder, `.dat`, in the directory specified. Similar to git, this folder stores information about that Dat. File metadata and the dat link are stored to make it easy to resume sharing or downloading the same directory.
```
Comment (karissa) [inline docs/api.md]: maybe use `dat-link` here instead of 'dat link'

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Dat is a decentralized data tool for distributing data small and large.
+
+### Features:
+
+  * **Live sync** folders by sharing files as they are added to the folder.
+  * **Distribute large files** without copying data to a central server by connecting directly to peers.
+  * **Intelligently sync** by deduplicating data between versions.
+  * **Verify data integrity** using strong cryptographic hashes.
+  * **Work everywhere**, including in the [browser](https://github.com/karissa/hyperdrive-ui) and on the [desktop](https://github.com/juliangruber/dat-desktop).
```
Comment (karissa) [inline docs/index.md]: probably want to link to the dat.land repo here instead

File: README.md [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
 -------------- | ------------
 [![Build status](https://ci.appveyor.com/api/projects/status/github/maxogden/dat?branch=master&svg=true)](https://ci.appveyor.com/project/maxogden/dat) | [![Travis](https://api.travis-ci.org/maxogden/dat.svg)](https://travis-ci.org/maxogden/dat)
 
+### Development Status
 
-This is the Dat 1.0 pre-release candidate. We want to make Dat into a data versioning, forking, and syncing tool. The first feature set we are working on is easy file distribution.
+This is the Dat 1.0 release candidate (RC2). We are actively seeking feedback & developing this release candidate. Follow [this issue](https://github.com/datproject/projects/issues/5) for the Dat CLI road map discussion and see [known RC2 issues](https://github.com/maxogden/dat/issues/486).
 
-Key features:
+### Key features:
```
Comment (karissa) [inline README.md]: maybe we should symlink docs/index.md to be readme.md, so we don't have to copy things over and can just edit readme.md and that's the first thing people see when they go to the generated docs page

**Comment by clkao** [comment]
LGTM.

We should discuss how to organize docs with docs.dat-data.com afterwards, and move non-cli intro to the docs repo

**Pull request merged**